packages/rulesets/src/spectral/functions/tags-not-allowed-in-properties-bag.ts

```diff
@@ -0,0 +1,29 @@
+import _ from "lodash"
+import { deepFindObjectKeyPath, getProperties } from "./utils"
+
+const TAGS = "tags"
+const PROPERTIES = "properties"
+const NestedPROPERTIES = "properties"
+const ERROR_MESSAGE = "Tags should not be specified in the properties bag for proxy resources. Consider using a Tracked resource instead."
+
+export const tagsNotAllowedInPropertiesBag = (definition: any, _opts: any, ctx: any) => {
+  const properties = getProperties(definition)
+  const errors = []
+
+  if ("tags" in properties && !("location" in properties)) {
+    errors.push({
+      message: ERROR_MESSAGE,
+      path: _.concat(ctx.path, PROPERTIES, TAGS),
+    })
+  }
+
+  const deepPropertiesTags = deepFindObjectKeyPath(definition.properties.properties, TAGS)
+  if (deepPropertiesTags.length > 0) {
+    errors.push({
+      message: ERROR_MESSAGE,
+      path: _.concat(ctx.path, PROPERTIES, NestedPROPERTIES, deepPropertiesTags[0]),
```

Comment:
Does this work for the path? The doc comment for `deepFindObjectKeyPath` says it should return the full path
```suggestion
      path: deepPropertiesTags,
```

Reply:
I am sending only specified properties to the method which will not return the path we are expecting deepFindObjectKeyPath(definition.properties.properties, TAGS)
I also tested the suggestion you added this is resulting empty/wrong paths